docs/overview.md

```diff
@@ -0,0 +1,103 @@
+inventory data from a variety of external sources.
+
+### Job Templates
+A Job Template is a definition and set of parameters for running
+`ansible-playbook`.  If defines metadata about a given playbook run, such as:
+
+* a named identifier
+* an associated inventory to run against
+* the project and `.yml` playbook to run
+* a variety of other options which map directly to ansible-playbook
+  arguments (extra_vars, verbosity, forks, limit, etc...)
+
+### Credentials
+awx stores sensitive credential data which can be attached to `ansible-playbook`
+processes that it runs.  This data can be oriented towards SSH connection
+authentication (usernames, passwords, SSH keys and passphrases),
+ansible-specific prompts (such as Vault passwords), or environmental
+authentication values which various Ansible modules depend on (such as setting
+`AWS_ACCESS_KEY_ID` in an environment variable, or specifying
+`ansible_ssh_user` as an extra variable).
+
+## awx's Interaction with Ansible
+The touchpoints between awx and Ansible are mostly encompassed by
+everything that happens *after* a job is started in awx.  Specifically, this
+includes:
+
+* Any time a Job Template is launched
+* Any time a Project Update is performed
+* Any time an Inventory Sync is performed
+* Any time an Adhoc Command is run
+
+### Spawning Ansible Processes
+awx relies on a handful of stable interfaces in its interaction with Ansible.
+The first of these are the actual CLI for `ansible-playbook` and
+`ansible-inventory`.
+
+When a Job Template or Project Update is run in awx, an actual
+`ansible-playbook` command is composed and spawned in a pseudoterminal on one
+of the servers/containers that make up the awx installation.  This process runs
+until completion (or until a configurable timeout), and the return code,
+stdout, and stderr of the process are recorded in the awx database.  Adhoc
+commands work the same way, though they spawn `ansible` processes instead of
+`ansible-playbook`.
+
+Similarly, when an Inventory Sync runs, an actual `ansible-inventory` process
+runs, and its output is parsed and persisted into the awx database as Hosts and
+Groups.
+
+awx relies on stability in CLI behavior to function properly across Ansible
+releases; this includes the actual CLI arguments _and_ the behavior of task
+execution and prompts (such as password, become, and Vault prompts).
+
+### Capturing Event Data
+awx applies an Ansible callback plugin to all `ansible-playbook` and `ansible`
+processes it spawns.  This allows Ansible events to be captured and persisted
+into the awx database; this process is what drives the "streaming" web UI
+you'll see if you launch a job from the awx web interface and watch its results
+appears on the screen.  awx relies on stability in this plugin interface, the
+heirarchy of emitted events based on strategy, and _especially_ the structure
+of event data to work across Ansible releases.
+
```

Comment:
Event Hierarchy Linkage
![ansible callback events job events - event in tower 5](https://user-images.githubusercontent.com/722880/35641610-ae7f1dea-068e-11e8-84fb-0f96043d53e4.png)